src/d3/hapticplot.d3.spec.ts

```diff
@@ -53,3 +87,44 @@ function getPosition(element: HTMLElement, shape: string): Array<{x: number, y:
   }
   return positionArray;
 }
+
+// returns array of each generated objects color
+function getColor(element: HTMLElement, shape: string): Array<string>{
+  const childrenArray = element.querySelectorAll(shape);
+  const attrArray: Array<string> = [];
+  for (const child of (childrenArray as any)){
+    attrArray.push((child as any).getAttribute('color'));
+  }
+  return attrArray;
```

Comment:
Something cool you can do is 
```suggestion
  return Array.from(element.querySelectorAll(shape).map((point: 'type of point') => point.getAttribute('color'))
```
This is more idiomatic typescript; also avoid using `any`....I'm not sure what 'type of point' is but if you need help correctly annotating this let me know!
Same comment applies to all these helper methods.